src/js/preshuffle.s

```diff
@@ -756,17 +756,225 @@ MaybeSetCheckpoint:
+  .byte $86,$20,$20,$20 ; Ex _ _ _
+  .byte $20,$20,$1f     ;  _ _ |
+
+.org $bb1b
+  .byte $20             ;  _
+.org $bb1f ; clear out the experience count thats there initially
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $85,$20,$20,$20 ; Mp _ _ _
+  .byte $9d,$20,$20,$20 ;  / _ _ _
+
+
+
 .segment "1a", "1b", "fe", "ff"
 ;.bank $34000 $8000:$4000
 
 ;;; Numeric displays
 .org $8ee9  ; 06 - was LV(menu) but now it's difficulty
   .word (Difficulty)
-  .byte $3c,$2b,$03,$00 ; display right of lvl
+  .byte $56,$2b,$03,$00 ; display left of exp
 .org $8f19  ; 0e - was unused, now it's LV(menu)
   .word (PlayerLevel)
   .byte $29,$29,$03,$00 ; copied from $34ee9
 
+;; Move the position of the main UI number elements
+.org $8ec7 ; Lv
+  .byte $36,$2b
+.org $8ecd ; Money
+  .byte $3a,$2b
+.org $8ed3 ; Experience
+  .byte $5a,$2b
+; Experience to next level is now unused,
+; if you want to use it for something, make sure the drawing isn't skipped
+; .org $8ed9 ; Experience Cap
+;   .byte $5a,$2b
+
+.pushseg "13", "fe", "ff"
+;; Prevent EXP Left from being drawn
+;InitializeStatusBarNametable
+.org $bace
+  jsr DisplayNumberInternalButSkipExpToNextLevel
+.reloc
+DisplayNumberInternalButSkipExpToNextLevel:
+  cmp #$03
+  beq +
+  jmp DisplayNumberInternal
++ rts
+.popseg
+
+;;; HP / Force bar display
+.org $8d1f
+  lda #$23; Move the player HP bar left 3 places
+.org $8db2
+  lda #$43; Move the player force bar left 3 places
+
+;;; ----------------------------------------------------
+;;; KillObject
+;;; Recalculate PlayerEXP to count down from max instead of up
+
+.org $9152
+  ; Instead of calling AwardExperience immediately, just store the obj offset
+  ; and push it onto the stack for use later. The code needs to check for an
+  ; object replacement and we don't wanna clobber that
+  jmp PushObjOffsetToStack
+.assert * <= $9155
+
+.org $9155
+CheckForObjectReplacementOrDeathAnimation:
+
+.reloc
+PushObjOffsetToStack:
+  tya
+  pha
+  ; Return back to where we started without touching the stack
+  jmp CheckForObjectReplacementOrDeathAnimation
+
+;; Update the level up check
+.org $916a
+  pla
+  tay
+  lda PlayerLevel
+  and #$f0
+  beq +
+    jmp ExitWithoutDrawingEXP
++ jsr AwardExperiencePoints
+  ; carry clear means we leveled up
+  bcc LevelUp
+  ; but it doesn't check if we were exactly at zero EXP so check for that now
+  lda PlayerExp
+  ora PlayerExp+1
+  cmp #$00
+  beq LevelUp
+  jmp UpdateCurrentEXPAndExit
+LevelUp:
+  ; double check here that we aren't already max level
+  inc PlayerLevel
+  lda PlayerLevel
+  asl  ; note: clears carry
+  tay
+  lda PlayerExp
+  adc $8b9e,y      ; NextLevelExpByLevel
+  sta PlayerExp
+  lda PlayerExp+1
+  adc $8b9f,y
+  sta PlayerExp+1
+  jmp UpdatePlayerMaxHPAndMPAfterLevelUp
+.assert * <= $91c7
+FREE_UNTIL $91c7
+
+
+.reloc
+UpdatePlayerMaxHPAndMPAfterLevelUp:
```

Comment:
This should be doable with a `.move $918b, 43` or `.move $9188, 46` (if you want to include the `inc`) and then we don't need to re-copy the whole thing?

I might take a stab at reworking some of this afterwards...